test/cases/generate-critical-css/generate-critical-css.spec.js

```diff
@@ -5,54 +5,71 @@ const { JSDOM } = jsdom;
+    before((done) => {
       webpack(webpackConfig, () => {
-        const indexHtmlString = fs.readFileSync(`${buildDirectory}/index.html`, { encoding: 'utf8' });
-        const indexHtmlDom = new JSDOM(indexHtmlString);
-        const inlineStyleTags = indexHtmlDom.window.document.querySelectorAll('style');
-        
-        assert.equal(inlineStyleTags.length, 1);
-        assert.equal(inlineStyleTags[0].getAttribute('type'), 'text/css');
+        indexHtmlString = fs.readFileSync(`${buildDirectory}/index.html`, { encoding: 'utf8' });
+        indexHtmlDom = new JSDOM(indexHtmlString);
 
         done();
       });
     });
 
-    it('should generate the expected critical <link> tag', (done) => {
-      webpack(webpackConfig, () => {
-        const indexHtmlString = fs.readFileSync(`${buildDirectory}/index.html`, { encoding: 'utf8' });
-        const indexHtmlDom = new JSDOM(indexHtmlString);
-        const linkTags = indexHtmlDom.window.document.querySelectorAll('link');
-        
-        assert.equal(linkTags.length, 2);
-        assert.equal(linkTags[0].getAttribute('rel'), 'preload');
-        assert.equal(linkTags[0].getAttribute('as'), 'style');
+    it('should generate the expected critical inline <style> tag', () => {
+      const inlineStyleTags = indexHtmlDom.window.document.querySelectorAll('style');
 
-        assert.equal(linkTags[1].getAttribute('rel'), 'preload');
-        assert.equal(linkTags[1].getAttribute('as'), 'style');
+      assert.equal(inlineStyleTags.length, 1);
+      assert.equal(inlineStyleTags[0].getAttribute('type'), 'text/css');
+    });
 
-        done();
-      });
+    it('should generate the expected critical <link> tag', () => {
+      const linkTags = indexHtmlDom.window.document.querySelectorAll('link');
+      
+      assert.equal(linkTags.length, 2);
+      assert.equal(linkTags[0].getAttribute('rel'), 'preload');
+      assert.equal(linkTags[0].getAttribute('as'), 'style');
+
+      assert.equal(linkTags[1].getAttribute('rel'), 'preload');
+      assert.equal(linkTags[1].getAttribute('as'), 'style');
     });
 
-    it('should generate the expected critical <noscript> tag', (done) => {
-      webpack(webpackConfig, () => {
-        const indexHtmlString = fs.readFileSync(`${buildDirectory}/index.html`, { encoding: 'utf8' });
-        const indexHtmlDom = new JSDOM(indexHtmlString);
-        const noscriptTags = indexHtmlDom.window.document.querySelectorAll('noscript');
-        
-        assert.equal(noscriptTags.length, 2);
+    it('should generate the expected critical <noscript> tag', () => {
+      const noscriptTags = indexHtmlDom.window.document.querySelectorAll('noscript');
+      
+      assert.equal(noscriptTags.length, 2);
+    });
+
+  });
+
+  describe('public path', () => {
+    let indexHtmlString;
+    let indexHtmlDom;
 
+    const config = merge({ output: { publicPath: '//cdn.example.com/' } }, webpackConfig);
```

Comment:
instead of pulling in a dependency just for this (**webpack-merge**) would it make sense to use `Object.assign` instead maybe?

```javascript

const config = Object.assign({ output: { publicPath: '//cdn.example.com/' } }, webpackConfig);
```

Curious if it would provide the same output we're looking for, just minus the added dependency?

cc: @anthonygore

Reply:
good point.